### ISSUE TITLE
Important things to keep in mind

### ISSUE THREAD

**Issue by andrerpena** [matterspace/matterspace#23]
Feature degradation
---

Our biggest competitor is UserVoice which only provides 1 of our features. Our closer competitors are UseResponse and Helprace which offers a lot more features (just like us) and they are quite smaller than UserVoice. But also, it doesn't seem like offering less features is necessarily better too. Wantoo is more like UserVoice and they are of the same size as UseResponse and Helprace. So, should we provide more or less features? **We should be flexible and allow the users to select which feature they want and adapt**.

If we offer **less features**, we risk not retaining customers that would need more features.

If we offer **more features**, we may get customers wanting more, but also we're at an ever bigger risk. We basically plan to offer feedback management, issue tracking and knowledge base. Chances are most of our potential customers already have at least one of these features using a competitor product (UserVoice, YouTrack, Jira, Trello). For that reason, forcing them into a product that would require them to migrate would be risky.

For that reason, we have to allow our customers to **select which features they want**.

On the surface it seems quite easy, because we just need to **hide the tab**, like getting rid of the Issues tab. Quite simple. Except it's not.

If the user disables the Issues tab, we can't just go ahead and delete all issues. That would be too dangerous. What we actually have to do is to hide the tab, keeping all previously registered threads intact. But that require some extra work:

- On the thread autocomplete (when the user types # in the MD editor), we now should not display Issues.
- Releases are associated with multiple threads, we should now stop displaying Issues, even though there might be associated issues.
- Issues should stop being displayed in the timeline.

**TL;DR:** We have to implement everything keeping in mind that the user might disable certain features.
